spec/jobs/subscription_placement_job_spec.rb

```diff
@@ -217,6 +217,18 @@
         end
       end
     end
+
+    context "when the proxy order fails to generate an order" do
+      before do
+        allow(proxy_order).to receive(:initialise_order!) { nil }
```

Comment:
are we really sure `#initialise_order!` will return nil on failure? I see a few cases where it would raise an exception so we might not be testing a production-like scenario in this test.

Reply:
I updated it to be more specific and handle the case where the proxy order's `order` is `nil`.